client/material.ts

```diff
@@ -23,20 +24,32 @@ function mapPiece(piece: string, variant: string): string {
         if (piece === '+p' || piece === 'pp') return 'c';
         return piece;
     }
+    if (variant == 'chak') {
+        if (piece === '+k' || piece === 'pk') return 'k';
+        return piece;
+    }
     return piece;
 }
 
 export function calculateInitialImbalance(variant: Variant): MaterialImbalance {
     let imbalances : MaterialImbalance = {};
     for (let piece of variant.pieceRoles('white')) imbalances[mapPiece(piece, variant.name)] = 0;
     for (let piece of variant.pieceRoles('black')) imbalances[mapPiece(piece, variant.name)] = 0;
+    if (variant.promotion === 'shogi') {
+        for (let piece of variant.promoteablePieces) {
+            imbalances[mapPiece('p' + piece, variant.name)] = 0;
```

Comment:
Yes, I know there is a promotedRole function in chess.ts, but it takes a parameter of cg.Piece type, which would require calling it like "promotedRole(variant, {role: piece + '-piece', color: 'white' , promoted: false} as cg.Piece)", which I feel is really cluttered